51.md

```diff
@@ -108,3 +109,7 @@ An event with kind `30000` is defined as a parameterized replaceable list event
 ### Categorized Bookmarks List
 
 An event with kind `30001` is defined as a parameterized replaceable list event for categorizing bookmarks. The 'd' parameter for this event holds the category name of the list. Any standardized tag can be included in a Categorized Bookmarks List.
+
+### Categorized Reference List
+  
+An event with kind `30303` is defined as a parameterized replaceable list event for categorizing user references. The 'd' parameter of this event holds the category name of the list. The reference lists may have metadata tags such as 'title', 'image', 'summary' as defined in [NIP-23 - Long-form Content](23.md) . Items in these lists are `r` tags, and follow the schema of Kind 3 events as defined in [NIP-02 - Contact List and Petnames](02.md). Each `r` tag entry should contain the 'reference URL' and the 'description', `["r", <reference URL>, <description>]`. Any standard tag can be included in a reference list.
```

Comment:
"categorizing user references" sounds like `p` tags to me. But this is for link trees right? So it could be `e` tags, `p` tags, urls, etc right?

Reply:
I used user references alluding to the 'r' tag, can you think of a better name? 'Categorized Links/URIs List'? And yes, you could use 'e', 'p', 'a' or other standardized tags to be included.

Reply:
I think this should just be a `kind:30001` tbh, it is in fact a list of bookmarks, right? Perhaps you can make a 30001 `d`ed `"linktree"` or something like that to tag the `kind:30001`s you want to display within this use case, or just display in this client those `kind:30001`s that are composed solely of `r` tags.

I don't think this warrants a specific kind especially now that kinds are [more scarce](https://github.com/nostr-protocol/nips/blob/master/01.md).

Reply:
I constantly add `title`, `description`, `image` to `kind:30001` -- don't let the fact that it's not explicitly named as such deter you from using/adding those.

I think the most cross-compatible, sensible way for this use case is to just create these as 30001s, and, in your client, only display those kind:30001s that have the data you want (e.g. `title`, `description`, `d` and the rest just `r` tags.

If you wanted a single query to find the users who have set this use case up you could `l` label the events with the name of the use case (e.g. linktree)

Reply:
`l`, maybe even with a pubkey you control is definitely a workable solution.

Reply:
I mean, yes, it is definitely a workable solution, kind 30001 with `l` tags. However, I still think that if we consider kind 30001 as a kind for everything we can think of, it might not be so cross-compatible in the future, because its parsing or visualisation might be very varied. Of course, this is not a problem for specific uses, because everyone will know what to do with the content of their 30001s. But we already have nip78 for that, don't we? For example, currently listr, one of the most focused list creation and management clients, does not allow you to add anything that is not a valid nip19 identifier, and type 30001 does not have a standardised `["tag", <reference URL>, <description>]`' scheme. This was my reasoning behind kind 30303.

Reply:
The labels is a good point. I wish tags were more extensible, the positional argument thing is so brittle. But if we specified that you can always add a display value (similar to petnames) at the end of any of these target tags would that make 30001 work?

Reply:
Yes, it would work. From my point of view there are two possibilities, either we rework the definition of kind 30001 to add this schema with petnames, the optional metadata fields, and the possibility to add more than just nip19 identifiers, or we add kind 30303 as a list of external references to nostr, more flexible in tags and content.
For me, adding kind 30303 has several advantages, such as a clearer and cleaner design, and defining more specific use cases for applications that reference external content. I'm also thinking about backwards compatibility and whether reworking kind 30001 might 'break' other clients that have been using it for a long time.
In any case, I will be happy with any solution. My purpose was to push forward the standardisation of a need that I have already seen in various clients that have created lists with random kinds (e.g. Pinstr with kind 33888, zap.stream with kind 37777 for maps), and for the purpose of my own client.
On the other hand, I don't understand the argument about the scarcity of kinds when parameterised replaceable events are 30000 <= n < 40000 and we have only standardised 19 of them, leaving 39980 possibilities to define more specific use cases.
I would like to hear more points of view here. @fiatjaf @verbiricha @sepehr-safari

Reply:
I am also in favour of a more generic list type like kind 30001. That would allow us to not only point to external urls but also to kind 1063 (NIP-94) events. I would be good to be able to mix different types of references in one list.

Reply:
I would use a `30001` as well. But instead of centering the website on npubs (which forces you to deal with all the other lists), Nostree should display by naddr. The website should point to individual lists, not to all lists of an author. 

As an example: if I want to do an Amethyst list, I can do a `30001:mykey:Amethyst` and share that around. I can even get `amethyst.social` to redirect to that specific list. I am still the author, so my picture will still be there as it is today. It's just not going to show all my other lists in the home page. 

Then later we can find a way to "pin" lists for each application.

Reply:
Well, I'll use kind30001 then, with 'r' tags and kind2 scheme for petnames. In a little while I will commit to add this to nip51. 
@vitorpamplona , yes, definitely, urls for individual lists is in the roadmap, I've slowed down the pace of development because I was waiting to clarify this kind issue better so I don't mess up the current user base too much. I will propose an easy way to migrate lists to kind30001.

Reply:
Just pushed the commit updating kind 30001